engine/src/lib.rs

```diff
@@ -206,6 +206,10 @@ mod contract {
             sdk::panic_utf8(errors::ERR_NOT_ALLOWED_TOO_EARLY);
         }
         Runtime::self_deploy(&bytes_to_key(KeyPrefix::Config, CODE_KEY));
+        io.write_storage(
```

Comment:
Why not remove the element?

```rust
io.remove_storate(&bytes_to_key(KeyPrefix::Config, CODE_STAGE_KEY));
```

Reply:
Yeah this would be better

Reply:
It seems that the approach from #716 fixes the same issue but without touching the storage which is good.